File: rumahsehat/src/main/java/apap/ta/rumahsehat/model/TagihanModel.java
package apap.ta.rumahsehat.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import java.io.Serializable;
import java.time.LocalDateTime;

@AllArgsConstructor
@Getter
@NoArgsConstructor
@Setter
@Entity
@Table(name = "tagihan")
public class TagihanModel implements Serializable {

	@Id
    @Size(max = 255)
    @GenericGenerator(name = "tagihan_id", strategy = "apap.ta.rumahsehat.model.TagihanIdGenerator")
    @GeneratedValue(generator = "tagihan_id")
    @Column(name = "kode", nullable = false)
    private String kode;
	
    @NotNull
    @Column(nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime tanggalTerbuat;

    @Column(nullable = false)
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'HH:mm")
    private LocalDateTime tanggalBayar;

    @NotNull
    @Column(name = "isPaid", nullable = false)
    private Boolean isPaid;

    @NotNull
    @Column(name = "jumlah_tagihan", nullable = false)
    private Integer jumlahTagihan;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "kodeAppointment", referencedColumnName = "kode", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JsonIgnore
    private AppointmentModel appointment;
}

File: rumahsehat_mobile/lib/models/tagihan_model.dart
class Tagihan {
  final String kode;
  final bool isPaid;
  final String tanggalTerbuat;
  final String tanggalBayar;
  final int jumlahTagihan;

  const Tagihan(
      {required this.kode,
      required this.isPaid,
      required this.tanggalTerbuat,
      required this.tanggalBayar,
      required this.jumlahTagihan});

  factory Tagihan.fromJson(Map<String, dynamic> json) {
    String tanggalBayar;
    if (json['tanggalBayar'] != null) {
      tanggalBayar = json['tanggalBayar'];
    } else {
      tanggalBayar = '-';
    }
    return Tagihan(
        kode: json['kode'],
        isPaid: json['isPaid'],
        tanggalTerbuat: json['tanggalTerbuat'],
        tanggalBayar: tanggalBayar,
        jumlahTagihan: json['jumlahTagihan']);
  }
}
